### PR TITLE
add devcontainer configs and docker network orchestration

### PR DESCRIPTION
**Pull request by jimmoffet**
Provided some local orchestration that allows us to connect the api, worker, postgres and redis services within this repo to one another, as well as to the notifications-admin repo via docker containers. We are also making use of vscode devcontainers, which allow for hot-reloading the notify flask apps, without needing install locally, restart containers or re-establish networking.

Step to set up:

**NOTE:** You'll only need to create the docker network once to connect both repos, using: `docker network create notify-network`. You'll need to install (again, only once) the [Visual Studio Code Remote - Containers extension](https://code.visualstudio.com/docs/remote/containers) to make use of the devcontainer sugar. 

**NOTE:** If you make a change to the Dockerfile, make sure to rebuild the correlated docker image. If you make a change to the docker-compose.devcontainer.yml file, _or the .env file_, make sure to rebuild the correlated container. VScode _should_ prompt you with an actionable notification asking to rebuild the containers when it detects one of these changes, but it does not always do so reliably/predictably (it's also easy to miss these temporary toast notifications).

**NOTE:** The next step will fail without an .env file present at the top level of this repo ⚠️ See sample.env for details. If you ran without .env and it failed, see the note above.

Once installed, you can use `command+shift+p` to open the command pallet and select **Remote-Containers: Open Folder in Container**, then select the `devcontainer-api` directory.

Vscode will build the container and establish networking. I recommend selecting "show logs" in the bottom-right. You can unhide hidden toast messages about the build by clicking the notifications bell in the far bottom-right. Once the container has finished building, you should see a "ports" tab in the terminal window (cmd-j) and it should show at least one mapped port for 6011.

Open a new terminal instance with the plus sign and enter `make run-flask`. At this point, you should be able to access the api's status endpoint on the "/" route at `0.0.0.0:6011`

You can now open another terminal and run the celery worker with 'make run-celery', you should see a log message like `celery@cf2c1e85f7a6 ready` when the celery worker becomes available and is watching the queues.

At this point, the notifications-api should be fully functional 🎉

### CODE REVIEW

File: app/config.py [hunk 1]
```diff
@@ -100,14 +99,19 @@ class Config(object):
     DANGEROUS_SALT = os.getenv('DANGEROUS_SALT')
 
     # DB conection string
-    SQLALCHEMY_DATABASE_URI = os.getenv('SQLALCHEMY_DATABASE_URI')
+    SQLALCHEMY_DATABASE_URI = os.getenv(
+        'SQLALCHEMY_DATABASE_URI',
+        'postgresql://postgres:chummy@db:5432/notification_api',
+    )
```
Comment (jimmoffet) [inline app/config.py:105]: not just formatting ^

File: app/config.py [hunk 2]
```diff
@@ -428,10 +462,13 @@ class Development(Config):
 
     NOTIFY_ENVIRONMENT = 'development'
     NOTIFY_LOG_PATH = 'application.log'
-    NOTIFY_EMAIL_DOMAIN = "notify.tools"
+    NOTIFY_EMAIL_DOMAIN = os.getenv('NOTIFY_EMAIL_DOMAIN', 'dispostable.com')
```
Comment (jimmoffet) [inline app/config.py:466]: mod this

File: app/config.py [hunk 3]
```diff
@@ -466,17 +503,20 @@ class Test(Development):
     LETTER_SANITISE_BUCKET_NAME = 'test-letters-sanitise'
 
     # this is overriden in jenkins and on cloudfoundry
-    SQLALCHEMY_DATABASE_URI = os.getenv('SQLALCHEMY_DATABASE_URI', 'postgresql://localhost/test_notification_api')
+    SQLALCHEMY_DATABASE_URI = os.getenv(
+        'SQLALCHEMY_DATABASE_URI',
+        'postgresql://postgres:chummy@db:5432/notification_api',
+    )
```
Comment (jimmoffet) [inline app/config.py:510]: mod this

**Review by hartsick — APPROVED** [review]
We paired through setting this up locally, and were able to get a development environment working on both machines

**Pull request merged**